### PR TITLE
Shared graph spec

### PR DESCRIPTION
**Pull request by stuartnelson3**
fixes bug where globalConfig endtime/range squash individually set widget endtime/range on initial load.

@juliusv

### CODE REVIEW

File: app/assets/javascripts/angular/controllers/dashboard_ctrl.js [hunk 1]
```diff
@@ -87,11 +87,22 @@ angular.module("Prometheus.controllers")
...
+  $scope.originalEndTime = $scope.globalConfig.endTime;
+  $scope.originalRange = $scope.globalConfig.range;
   $document.keydown(function(ev) {
     var Z_KEY = 90;
     var ESC_KEY = 27;
     var validElement = ["INPUT", "TEXTAREA"].indexOf(document.activeElement.tagName) === -1;
     if (ev.keyCode === Z_KEY && validElement) {
+      $timeout(function() {
+        // If we have reached the initial load state, sets the
+        // globalConfig to its original state for cause a graph
```
Comment (juliusv) [inline app/assets/javascripts/angular/controllers/dashboard_ctrl.js]: "for cause"? "to cause"?

File: app/assets/javascripts/angular/controllers/dashboard_ctrl.js [hunk 2]
```diff
@@ -87,11 +87,22 @@ angular.module("Prometheus.controllers")
...
+        // refresh.
+        if ($.isEmptyObject($location.search())) {
+          $scope.globalConfig.endTime = $scope.originalEndTime;
+          $scope.globalConfig.range = $scope.originalRange;
+        }
+      })
```
Comment (juliusv) [inline app/assets/javascripts/angular/controllers/dashboard_ctrl.js]: missing semi

File: app/assets/javascripts/angular/controllers/dashboard_ctrl.js (hunk 1 above)
Comment (stuartnelson3) [inline app/assets/javascripts/angular/controllers/dashboard_ctrl.js]: hah yep, looks like my brain is dying

**Comment by grobie** [comment]
:+1: I want mixed times back :)

**Comment by juliusv** [comment]
:+1:

**Pull request merged**